src/goesproc/gradient.h

```diff
@@ -0,0 +1,170 @@
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+
+struct GradientPoint {
+  float units = 0;
+  float rgb[3];
+  float hsv[3];
+
+  GradientPoint fromRGB(float r, float g, float b) const {
+    float min, max;
+    GradientPoint out;
+    out.units = units;
```

Comment:
Looks like the only thing carried over from the original gradient point is the units field? It looks like it is only carried forward in the interpolate function but there it needs to be overwritten anyway. I think we can treat GradientPoint as const itself, as in, return as const. Then to construct one you need to specify RGB/HSV as well as the unit value.